internal/app/networkd/pkg/nic/nic.go

```diff
@@ -2,75 +2,323 @@
 	// Ensure we have an interface name defined
 	if iface.Name == "" {
-		result = multierror.Append(errors.New("interface must have a name"))
+		result = multierror.Append(result, errors.New("interface must have a name"))
 	}
 
 	// If no addressing methods have been configured, default to DHCP
 	// TODO: do we want this behavior or to be explicit with config
 	// so we dont configure every interface be default?
-	if len(iface.AddressMethod) == 0 {
-		iface.AddressMethod = append(iface.AddressMethod, &address.DHCP{NetIf: link})
+
+	// TODO readd this later when we've refactored address to not require a link here?
+	/*
+		if len(iface.AddressMethod) == 0 {
+			iface.AddressMethod = append(iface.AddressMethod, &address.DHCP{NetIf: link})
+		}
+	*/
+
+	// Handle netlink connection
+	conn, err := rtnl.Dial(nil)
+	if err != nil {
+		result = multierror.Append(result, err)
+		return nil, result.ErrorOrNil()
 	}
+	iface.rtnlConn = conn
+
+	// Need rtnetlink for MTU setting
+	// TODO: possible rtnl enhancement
+	nlConn, err := rtnetlink.Dial(nil)
+	if err != nil {
+		result = multierror.Append(result, err)
+		return nil, result.ErrorOrNil()
+	}
+	iface.rtConn = nlConn
 
 	return iface, result.ErrorOrNil()
 }
+
+// IsIgnored checks the network interface to see if it should be ignored and not configured
+func (n *NetworkInterface) IsIgnored() bool {
+	if n.Ignore || kernel.ProcCmdline().Get(constants.KernelParamNetworkInterfaceIgnore).Contains(n.Name) {
+		return true
+	}
+
+	return false
+}
+
+// Create creates the underlying link if it does not already exist
+func (n *NetworkInterface) Create() error {
+	log.Println("nic.create")
+	iface, err := net.InterfaceByName(n.Name)
+	if err == nil {
+		n.Link = iface
+		return err
+	}
+
+	log.Println("creating link")
+	if err = n.createLink(); err != nil {
+		log.Println("failed to create link")
+		return err
+	}
+
+	log.Println("finding link")
+	iface, err = net.InterfaceByName(n.Name)
+	if err != nil {
+		log.Println("failed to find link")
+		return err
+	}
+	log.Println("found link", iface)
+	n.Link = iface
+
+	return nil
+}
+
+func (n *NetworkInterface) Configure() (err error) {
+	// Ensure link is down if we should ignore the link
+	// so we are at an appropriate default state
+	if n.IsIgnored() {
+		log.Println("ignored", n.Link.Name)
+		return n.rtnlConn.LinkDown(n.Link)
+	}
+
+	// Would be nice to set MTU here, but since we've deferred getting that
+	// information till the Addressing portion, we'll do it there
+
+	if n.Bonded {
+		if err = n.configureBond(n.Link.Index, n.BondSettings); err != nil {
+			return err
+		}
+		// Surely there's a better way to do a *uint32
+		var bondIndex *uint32
+		bondIndex = new(uint32)
+		*bondIndex = uint32(n.Link.Index)
+
+		if err = n.enslaveLink(bondIndex, n.SubInterfaces...); err != nil {
+			return err
+		}
+	}
+
+	log.Println("bringing up", n.Link.Name)
+	if err = n.rtnlConn.LinkUp(n.Link); err != nil {
+		log.Printf("failed to bring up %s: %v", n.Name, err)
```

Comment:
Return the error only please. It should be up to the consumer to print the error if they please.